OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs

```diff
@@ -766,7 +766,13 @@ static bool Kick(S server, Connection conn, Session.Client client, string s)
 				}
 
 				var kickClient = server.GetClient(kickConn);
-				if (server.State == ServerState.GameStarted && !kickClient.IsObserver)
+
+				var ws = server.worldPlayers
+					.Where(p => p != null && p.ClientIndex == kickClient.Index)
+					.Select(p => p.Outcome)
+					.Single();
```

Comment:
Are you sure this doesn't crash when trying to kick a spectator (which has no player associated I assume)?